protx/utils/maltreatment.py

```diff
@@ -102,13 +110,96 @@ def maltrt_stacked_bar(maltrt_data_dict):
 def maltreatment_plot_figure(area, geoid, variables, unit):
     logger.info("Selected maltreatment variables are: {}".format(variables))
     user_select_data = {
-        'area': area,
-        'geoid': geoid,
-        'units': unit,
-        'variables': ','.join(['"{}"'.format(v) for v in variables])
+        "area": area,
+        "geoid": geoid,
+        "units": unit,
+        "variables": ",".join(['"{}"'.format(v) for v in variables]),
     }
     db_conn = sqlite3.connect(db.cooks_db)
     maltrt_data = query_return(user_select_data, db_conn)
     db_conn.close()
     plot_figure = maltrt_stacked_bar(maltrt_data)
     return json.loads(plot_figure.to_json())
+
+
+def get_child_mal_by_age_charts(area, geoid):
+    db_conn = sqlite3.connect(db.cooks_db)
+    light_green_to_blue_color_palette = [
+        "#eff5d6",
+        "#c6e8b0",
+        "#8fcca1",
+        "#62ad9c",
+        "#3c7d8a",
+        "#26547a",
+        "#eff5d6",
+        "#c6e8b0",
+        "#8fcca1",
+        "#62ad9c",
+    ]
+
+    query = """
+    SELECT YEAR, {age_variables} as AGE
+        from maltreatment m
+    where m.GEOID = "{geoid}" and
+        m.MALTREATMENT_NAME= "{maltreatment_name}" and
+        m.GEOTYPE = "{area}" and
+        m.UNITS = "{units}"
+          GROUP BY YEAR;
+    """
+    age_variables = [
+        "VALUE_UNDER_FIVE",
+        "VALUE_FIVE_TO_NINE",
+        "VALUE_TEN_TO_FOURTEEN",
+        "VALUE_OVER_FOURTEEN",
+    ]
+    selection = {
+        "units": "percent",
+        "area": area,
+        "geoid": geoid,
+        "maltreatment_name": "ALL",
+        "age_variables": ",".join([f"m.{v}" for v in age_variables]),
+    }
+    maltrt_age_data_frame = pd.read_sql_query(query.format(**selection), db_conn)
+    age_labels = ["0-4", "5-9", "10-14", "15-17"]
+    # [2:7] to only get the values for each age group from the query
+    age_data = {row.YEAR: row[2:7] for row in maltrt_age_data_frame.itertuples()}
+
+    # Normalize the data for each age group
+    sum_per_year = {year: sum(age_data[year]) for year in age_data}
+    normalized_age_data = {
+        year: [value / sum_per_year[year] for value in age_data[year]]
+        for year in age_data
+    }
+
+    data = []
+    for i, age in enumerate(age_labels):
+        # TODO
```

Comment:
maybe drop these TODO notes here and in 191

Reply:
I addressed the error I wanted to catch and updated with latest push. Thank you!